backend/btrixcloud/migrations/migration_0037_upload_pages.py

```diff
@@ -19,54 +18,56 @@ def __init__(self, mdb, **kwargs):
 
         self.background_job_ops = kwargs.get("background_job_ops")
         self.page_ops = kwargs.get("page_ops")
-
-    async def org_upload_pages_already_added(self, oid: UUID) -> bool:
-        """Check if upload pages have already been added for this org"""
-        if self.page_ops is None:
-            print(
-                f"page_ops missing, assuming pages need to be added for org {oid}",
-                flush=True,
-            )
-            return False
-
-        mdb_crawls = self.mdb["crawls"]
-        async for upload in mdb_crawls.find({"oid": oid, "type": "upload"}):
-            upload_id = upload["_id"]
-            _, total = await self.page_ops.list_pages(upload_id)
-            if total > 0:
-                return True
-        return False
+        self.coll_ops = kwargs.get("coll_ops")
 
     async def migrate_up(self):
         """Perform migration up.
 
         Start background jobs to parse uploads and add their pages to db
         """
-        if self.background_job_ops is None:
-            print(
-                "Unable to start background job, missing background_job_ops", flush=True
-            )
+        if not self.background_job_ops or not self.page_ops or not self.coll_ops:
+            print("Unable to start migration, missing ops", flush=True)
             return
 
         mdb_orgs = self.mdb["organizations"]
-        async for org in mdb_orgs.find():
-            oid = org["_id"]
+        mdb_crawls = self.mdb["crawls"]
+
+        async for org_dict in mdb_orgs.find():
+            org = Organization.from_dict(org_dict)
```

Comment:
Is there any reason to bin this by org, instead of just find all uploads? That way could show a running total of N out of M uploads?

Reply:
In part to be able to gauge progress, in part because the post-page collection updates that need to happen are easier to do per-org. But also a bit of a remnant of how this migration used to work, can change if you want.

Reply:
Ahh right, the per-org operations are at the end. I think its fine, there is progress already shown per org.